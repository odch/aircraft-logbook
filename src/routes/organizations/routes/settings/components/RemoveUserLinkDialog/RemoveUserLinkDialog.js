import React from 'react'
import PropTypes from 'prop-types'
import { injectIntl, FormattedMessage } from 'react-intl'
import { withStyles } from '@material-ui/core/styles'
import Button from '@material-ui/core/Button'
import Dialog from '@material-ui/core/Dialog'
import DialogActions from '@material-ui/core/DialogActions'
import DialogContent from '@material-ui/core/DialogContent'
import DialogContentText from '@material-ui/core/DialogContentText'
import DialogTitle from '@material-ui/core/DialogTitle'
import SaveButton from '../../../../../../components/SaveButton'
import {
  member as memberShape,
  intl as intlShape
} from '../../../../../../shapes'

const MemberAttribute = props => (
  <DialogContentText className={props.className}>
    <FormattedMessage
      id={`organization.member.removeuserlink.dialog.${props.label}`}
    />
    :&nbsp;
    {props.value}
  </DialogContentText>
)

MemberAttribute.propTypes = {
  label: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  className: PropTypes.string
}

const styles = {
  attributesWrapper: {
    marginTop: '1em'
  },
  attribute: {
    margin: 1
  }
}

class RemoveUserLinkDialog extends React.Component {
  handleClose = () => {
    if (!this.props.submitting && this.props.onClose) {
      this.props.onClose()
    }
  }

  render() {
    const { member, submitting, classes, onConfirm } = this.props
    return (
      <Dialog
        onClose={this.handleClose}
        data-cy="remove-member-link-dialog"
        open
      >
        <DialogTitle>
          <FormattedMessage id="organization.member.removeuserlink.dialog.title" />
        </DialogTitle>
        <DialogContent>
          <DialogContentText>
            <FormattedMessage id="organization.member.removeuserlink.dialog.text" />
          </DialogContentText>
          <div className={classes.attributesWrapper}>
            {member.firstname && (
              <MemberAttribute
                label="firstname"
                value={member.firstname}
                className={classes.attribute}
              />
            )}
            {member.lastname && (
              <MemberAttribute
                label="lastname"
                value={member.lastname}
                className={classes.attribute}
              />
            )}
            {member.nr && (
              <MemberAttribute
                label="nr"
                value={member.nr}
                className={classes.attribute}
              />
            )}
          </div>
        </DialogContent>
        <DialogActions>
          <Button onClick={this.handleClose} disabled={submitting}>
            <FormattedMessage id="organization.member.removeuserlink.dialog.buttons.cancel" />
          </Button>
          <SaveButton
            onClick={onConfirm}
            color="secondary"
            variant="contained"
            label={this.props.intl.formatMessage({
              id:
                'organization.member.removeuserlink.dialog.buttons.removeuserlink'
            })}
            data-cy="remove-user-link-button"
            disabled={submitting}
            inProgress={submitting}
          />
        </DialogActions>
      </Dialog>
    )
  }
}

RemoveUserLinkDialog.propTypes = {
  organizationId: PropTypes.string.isRequired,
  member: memberShape.isRequired,
  submitting: PropTypes.bool,
  classes: PropTypes.object.isRequired,
  onConfirm: PropTypes.func,
  onClose: PropTypes.func,
  intl: intlShape
}

export default withStyles(styles)(injectIntl(RemoveUserLinkDialog))
