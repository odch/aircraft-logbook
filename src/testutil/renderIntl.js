import React from 'react'
import renderer from 'react-test-renderer'
import { createMount } from '@material-ui/core/test-utils'
import { IntlProvider } from 'react-intl'
import pretty from 'pretty'
import messages from '../messages'

const LOCALE = 'de'

const renderInIntlProvider = component => (
  <IntlProvider locale={LOCALE} messages={messages[LOCALE]}>
    {component}
  </IntlProvider>
)

const renderIntl = (component, options) =>
  renderer.create(renderInIntlProvider(component), options)

export const renderIntlUpdate = (renderer, component, options) =>
  renderer.update(renderInIntlProvider(component), options)

export const renderIntlMaterial = (component, returnPrettyHtml = false) => {
  const rendered = createMount()(renderInIntlProvider(component))
  return returnPrettyHtml ? pretty(rendered.html()) : rendered
}

export default renderIntl
