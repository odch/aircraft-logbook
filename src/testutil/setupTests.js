import * as enzyme from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import moment from 'moment'
import momentTimezone from 'moment-timezone'
import { init as initFeatureToggles } from '../util/featureToggles'

enzyme.configure({ adapter: new Adapter() })

moment.locale('de')
momentTimezone.locale('de')

initFeatureToggles()
