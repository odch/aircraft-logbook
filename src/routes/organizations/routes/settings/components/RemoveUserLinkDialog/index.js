import RemoveUserLinkDialog from './RemoveUserLinkDialog'

export default RemoveUserLinkDialog
