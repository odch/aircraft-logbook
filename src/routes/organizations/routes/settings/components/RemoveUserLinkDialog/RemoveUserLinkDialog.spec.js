import React from 'react'
import { renderIntlMaterial } from '../../../../../../testutil/renderIntl'
import RemoveUserLinkDialog from './RemoveUserLinkDialog'

describe('routes', () => {
  describe('organizations', () => {
    describe('routes', () => {
      describe('settings', () => {
        describe('components', () => {
          describe('RemoveUserLinkDialog', () => {
            it('renders correctly', () => {
              const member = {
                id: 'my_test_member',
                firstname: 'Max',
                lastname: 'Muster',
                nr: '23563'
              }
              const renderedValue = renderIntlMaterial(
                <RemoveUserLinkDialog
                  organizationId="my_org"
                  member={member}
                />,
                true
              )
              expect(renderedValue).toMatchSnapshot()
            })
          })
        })
      })
    })
  })
})
