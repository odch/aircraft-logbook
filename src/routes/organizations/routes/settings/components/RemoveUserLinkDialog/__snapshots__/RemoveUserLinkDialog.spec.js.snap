// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`routes organizations routes settings components RemoveUserLinkDialog renders correctly 1`] = `
"<div role=\\"presentation\\" class=\\"MuiDialog-root\\" data-cy=\\"remove-member-link-dialog\\" style=\\"position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px;\\">
  <div class=\\"MuiBackdrop-root\\" aria-hidden=\\"true\\" style=\\"opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;\\"></div>
  <div tabindex=\\"0\\" data-test=\\"sentinelStart\\"></div>
  <div class=\\"MuiDialog-container MuiDialog-scrollPaper\\" style=\\"opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;\\" role=\\"document\\" tabindex=\\"-1\\">
    <div class=\\"MuiPaper-root MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm MuiPaper-rounded\\" role=\\"dialog\\">
      <div class=\\"MuiDialogTitle-root\\">
        <h2 class=\\"MuiTypography-root MuiTypography-h6\\">Benutzer-Verknüpfung entfernen</h2>
      </div>
      <div class=\\"MuiDialogContent-root\\">
        <p class=\\"MuiTypography-root MuiDialogContentText-root MuiTypography-body1 MuiTypography-colorTextSecondary\\">Bist du sicher, dass du die Benutzer-Verknüpfung entfernen möchtest? Der verknüpfte Benutzer verliert damit den Zugriff auf diese Organisation. Du kannst den Benutzer später bei Bedarf jederzeit neu in die Organisation einladen.</p>
        <div class=\\"injectIntl(RemoveUserLinkDialog)-attributesWrapper-1\\">
          <p class=\\"MuiTypography-root MuiDialogContentText-root injectIntl(RemoveUserLinkDialog)-attribute-2 MuiTypography-body1 MuiTypography-colorTextSecondary\\">Vorname:&nbsp;Max</p>
          <p class=\\"MuiTypography-root MuiDialogContentText-root injectIntl(RemoveUserLinkDialog)-attribute-2 MuiTypography-body1 MuiTypography-colorTextSecondary\\">Nachname:&nbsp;Muster</p>
          <p class=\\"MuiTypography-root MuiDialogContentText-root injectIntl(RemoveUserLinkDialog)-attribute-2 MuiTypography-body1 MuiTypography-colorTextSecondary\\">Nummer:&nbsp;23563</p>
        </div>
      </div>
      <div class=\\"MuiDialogActions-root MuiDialogActions-spacing\\"><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-text\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiButton-label\\">Abbrechen</span><span class=\\"MuiTouchRipple-root\\"></span></button><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary\\" tabindex=\\"0\\" type=\\"submit\\" data-cy=\\"remove-user-link-button\\"><span class=\\"MuiButton-label\\">Benutzer-Verknüpfung entfernen</span><span class=\\"MuiTouchRipple-root\\"></span></button></div>
    </div>
  </div>
  <div tabindex=\\"0\\" data-test=\\"sentinelEnd\\"></div>
</div>"
`;
